bokeh/plotting/_renderer.py

```diff
@@ -316,6 +304,16 @@ def _process_sequence_literals(glyphclass, kwargs, source, is_user_source):
 
     return incompatible_literal_spec_values
 
+def _split_feature_trait(ft):
+    """Feature is up to first '_'. Ex. 'line_color' => ['line', 'color']"""
+    ft = ft.split('_', 1)
+    return ft if len(ft)==2 else ft+[None]
+
+def _is_visual(ft):
+    """Whether a feature trait name is visual"""
+    feature, trait = _split_feature_trait(ft)
+    return feature in ('line', 'fill', 'text', 'global') and trait is not None
```

Comment:
This is a pretty hacky way to check. In any case, should this include `'hatch'`? I'm willing believe it should not, but then the functions for handling color and alpha could use more specific, descriptive names.

Reply:
Answer revealed as yes in another issue/PR